### PR TITLE
handle unsaved changes on reboot

### LINKED ISSUE #1338: The "Restart chaiNer" button skips the unsaved data dialog
and results in loss of data.

### PR DESCRIPTION
**Pull request by stonerl**
fixes #1338

### CODE REVIEW

**Comment by stonerl** [comment]
CleanShot 2023-08-23 at 00 11 52

File: src/main/gui/main-window.ts [hunk 1]
```diff
@@ -190,38 +182,78 @@ const registerEventHandlerPostSetup = (
...
 
+    ipcMain.on('reboot-after-save', () => {
+        app.relaunch();
+        app.exit();
+    });
+
+    const handleUnsavedChanges = (
+        event: Electron.Event,
+        callback: () => void,
+        restartCallback?: () => void
```
Comment (joeyballentine) [inline src/main/gui/main-window.ts:200]: I feel like this should be more generic, like `onSave` and `onDontSave`

Comment (stonerl) [reply src/main/gui/main-window.ts]: done

File: src/main/gui/main-window.ts [hunk 2]
```diff
@@ -190,38 +182,78 @@ const registerEventHandlerPostSetup = (
...
+            // Save
+            event.preventDefault();
+            onSave();
+        }
+    };
+
+    ipcMain.handle('relaunch-application', (event) => {
+        if (backend.owned) {
+            backend.tryKill();
+        }
```
Comment (RunDevelopment) [inline src/main/gui/main-window.ts:230]: So when the user selects "Cancel", the backend will be killed the UI will stay open?

Comment (stonerl) [reply src/main/gui/main-window.ts]: Oh, thank you for pointing this out.

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**